src/Standards/Generic/Sniffs/ControlStructures/DisallowYodaConditionsSniff.php

```diff
@@ -144,14 +139,14 @@ public function isArrayStatic(File $phpcsFile, $arrayToken)
     {
         $tokens = $phpcsFile->getTokens();
 
-        $arrayEnd = null;
```

Comment:
See: https://github.com/PHPCSStandards/PHP_CodeSniffer/pull/446#discussion_r1573937663

Reply:
I have a niggly feeling (untested) that PHPStan (at a higher level than currently in use) would complain about it as the variable is not defined in _all_ branches of the if/else.

Reply:
I tested and I was not able to see this error. The error level that checks for undefined variables is level `1`. I tested with level `9` and still did not get any errors. I believe PHPStan understands the `return` inside the `else`. If I remove the `return`, it complains about the potentially undefined variable.

If you still prefer, I'd be happy to define `$end` to `0` before the `if` statement. But in that case, I guess I should also define `$start`?

Reply:
You know what, let's leave it. It's fine.